python/paddle/fluid/tests/unittests/test_recurrent_op.py

```diff
@@ -456,5 +456,145 @@ def create_rnn_op(self):
+            self.w2 = np.random.uniform(
+                -0.1, 0.1, size=(input_dim * 2, input_dim)).astype("float32")
+
+            self.emb = np.random.uniform(
+                -0.1, 0.1, size=(seq_len, batch_size,
+                                 input_dim)).astype("float32")
+
+            men_dim = (seq_len, batch_size, input_dim)
+            self.mems = np.zeros(shape=men_dim).astype("float32")
+            self.oy = np.matmul(self.emb, self.w1)
+
+        def step(self, step_id, x):
+            def dot_attention(query, memory):
+                attn = np.matmul(query, memory.transpose((0, 2, 1)))
+                weight = softmax(attn)
+                weight_memory = np.matmul(weight, memory)
+                return weight_memory, weight
+
+            def softmax(x):
+                return np.exp(x) / sum(np.exp(x))
+
+            if step_id == 0:
+                pre_mem = np.zeros_like(x)
+            else:
+                pre_mem = self.mems[step_id - 1]
+            concat_in = np.concatenate([x, pre_mem], 1)
+            new_mem = np.matmul(concat_in, self.w2)
+
+            new_mem = np.expand_dims(new_mem, 1)
+            new_mem, _ = dot_attention(new_mem, self.oy)
+            new_mem = np.squeeze(new_mem, 1)
+
+            self.mems[step_id] = new_mem
+            self.y[step_id] = self.mems[step_id]
+
+    input_dim = 2
+    batch_size = 3
+    sent_len = 3
+
+    def setUp(self):
+        self.setup_program()
+
+        self.data_field = {"x", "emb", "w1", "w2"}
+
+        self.input_shape = (self.sent_len, self.batch_size, self.input_dim)
+        self.output_shape = (self.sent_len, self.batch_size, self.input_dim)
+        self.py_rnn = RecurrentOpSubBlockTest.PySimpleRNN5(self.input_shape,
+                                                           self.output_shape)
+
+        with fluid.program_guard(self.main_program, self.startup_program):
+            rnn_out = self.create_rnn_op()
+            self.output = layers.mean(rnn_out)
+
+    def create_rnn_op(self):
+        x = layers.data(
+            shape=[self.sent_len, self.batch_size, self.input_dim],
+            dtype='float32',
+            name='x',
+            append_batch_size=False)
+        x.stop_gradient = False
+        x += 0.0
+
+        emb = layers.data(
+            name='emb',
+            shape=[self.sent_len, self.batch_size, self.input_dim],
+            dtype='float32',
+            append_batch_size=False)
+        emb.stop_gradient = False
+        emb += 0.0
+
+        w1 = layers.data(
+            shape=[self.input_dim, self.input_dim],
+            dtype='float32',
+            name='w1',
+            append_batch_size=False)
+        w1.stop_gradient = False
+        w2 = layers.data(
+            shape=[self.input_dim * 2, self.input_dim],
+            dtype='float32',
+            name='w2',
+            append_batch_size=False)
+        w2.stop_gradient = False
```

Comment:
It seems that not need to set stop_gradient here.

Reply:
why? w2 should also compute gradient